shared/src/data/GoodNumber.ts

```diff
@@ -149,6 +173,10 @@ export class GN {
     return a.lt(b) ? a : b;
   }
 
+  static areWithinNSigDigs(a: GN, b: GN, n: number): boolean {
+    return a.int.prec(n).eq(b.int.prec(n));
```

Comment:
Honestly didn't realize this `prec` function was a thing. To match Solidity more closely, one might argue we should be using it after every `mul` operation (since `Big.times` doesn't round automatically). Something to consider for future PRs.